assets/src/edit-story/app/media/media3p/api/media3pApiProvider.js

```diff
@@ -55,7 +55,18 @@ function Media3pApiProvider({ children }) {
   function getUrls(m) {
     if (m.type.toLowerCase() === 'image') {
       return Object.fromEntries(
-        new Map(m.imageUrls.map((u) => [u.imageName, u.url]))
+        new Map(
+          m.imageUrls.map((u) => [
+            u.imageName,
+            {
+              file: m.name,
+              height: u.height,
+              mime_type: u.mime_type,
+              source_url: u.url,
+              width: u.width,
```

Comment:
[nit]: Optional. Perhaps list in a logical order, with width and height together:

file,
source_url,
mime_type,
width,
height,